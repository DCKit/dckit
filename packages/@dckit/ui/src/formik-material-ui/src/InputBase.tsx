import * as React from 'react';
import MuiInputBase, {
  InputBaseProps as MuiInputBaseProps,
} from '@material-ui/core/InputBase';
import {
  FieldInputProps,
  FieldMetaProps,
  FieldHelperProps,
  useField,
  useFormikContext,
} from 'formik';

export interface InputBaseProps
  extends Omit<MuiInputBaseProps, 'name' | 'value' | 'error'> {
  name: string;
}

export function useFieldToInputBase<Val = unknown>(
  { disabled, name, ...props }: InputBaseProps,
  customize?: (
    props: [FieldInputProps<Val>, FieldMetaProps<Val>, FieldHelperProps<Val>]
  ) => Partial<Omit<InputBaseProps, 'name'>>
): MuiInputBaseProps {
  const { isSubmitting } = useFormikContext();
  const fieldProps = useField(name);

  const [field] = fieldProps;

  return {
    disabled: disabled ?? isSubmitting,
    ...props,
    ...field,
    ...customize?.(fieldProps),
  };
}

export function InputBase(props: InputBaseProps) {
  return <MuiInputBase {...useFieldToInputBase(props)} />;
}

InputBase.displayName = 'FormikMaterialUIInputBase';
