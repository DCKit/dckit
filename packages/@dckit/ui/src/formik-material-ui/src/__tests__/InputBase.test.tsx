import * as React from 'react';
import { Formik, Form } from 'formik';
import {
  render,
  fireEvent,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { InputBase } from '../InputBase';

test('InputBase Renders Correctly', () => {
  const { asFragment } = render(
    <Formik onSubmit={() => {}} initialValues={{ test: 'Input' }}>
      <Form>
        <InputBase name="test" />
      </Form>
    </Formik>
  );

  expect(asFragment()).toMatchSnapshot();
  // component.root.findByProps({ type: "text" }).props.onClick();
});

test('InputBase Renders Correctly disabled', () => {
  const { asFragment, getByTestId } = render(
    <Formik onSubmit={() => {}} initialValues={{ test: 'Input' }}>
      <Form>
        <InputBase name="test" inputProps={{ 'data-testid': 'input' }} />
      </Form>
    </Formik>
  );

  waitForElementToBeRemoved;

  fireEvent.change(getByTestId('input'), { target: { value: 'a' } });
  expect(asFragment()).toMatchSnapshot();
  // component.root.findByProps({ type: "text" }).props.onClick();
});
