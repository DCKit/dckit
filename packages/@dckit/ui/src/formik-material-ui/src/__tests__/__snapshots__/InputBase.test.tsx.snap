// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBase Renders Correctly 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiInputBase-root"
    >
      <input
        class="MuiInputBase-input"
        name="test"
        type="text"
        value="Input"
      />
    </div>
  </form>
</DocumentFragment>
`;

exports[`InputBase Renders Correctly disabled 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiInputBase-root"
    >
      <input
        class="MuiInputBase-input"
        data-testid="input"
        name="test"
        type="text"
        value="a"
      />
    </div>
  </form>
</DocumentFragment>
`;
